File: Personal Work For OCaml/Work_Shop_2.ml
(* https://ocaml.org/manual/5.2/lex.html#sss:stringliterals *)
(* syno: went over suffix encoding, continue to floats *)

(*   & & & & & & Commenting & & & & & &   *)

(* Single and 
multi line comments *)

(*
Nested
(* Comments within *)
Other comments
*)

(*   & & & & & & Literals & & & & & &   *)

(*   ^ ^ ^ ^ ^ Integers ^ ^ ^ ^ ^   *)

123;;
4_273;;
42_73;;
427_3 ;;
-34;;


(*  ^ ^ ^ ^ Integers: Type Modifiers ^ ^ ^ ^   *)

0xF5;;
0o12;;
0b110;;

(* No signed unsigned coding suffixes in python *) 
1l;;(*32int*) 
1L;;(*64int*)
1n;;(*nativeint*)

2;;
3.4;;


'h';;
(*int_of_char 'h';; to find out ASCII value*)
"Bye";;

false;;

(* ^ ^ ^ ^ ^ Different Radixes ^ ^ ^ ^ ^*)

(*x/o/b adding in the end of the literal, give examples*)

(*   & & & & & & Operators & & & & & &   *)

(*   & & & & & & String Formatting & & & & & &   *)

(*   & & & & & & Conditional Statements & & & & & &   *)

(*   & & & & & & Loops & & & & & &   *)

2 + 4;;
3 * 2;;
3.2 +. 2.0;; (* different operators for different types operators (float operators)*)
(1 < 2);;
"Hello " ^ " " ^ "World";;

File: FullStack/Personal Work For Rust/Work_Shop_2.rs
// https://www.tutorialspoint.com/rust/rust_data_types.htm
// Data Types

//   & & & & & & Comments & & & & & &   //

// Single line comment

/* 
Multi Line 
Comment
 */

//   & & & & & & Literals & & & & & &   // 
//   ^ ^ ^ ^ ^ Integers ^ ^ ^ ^ ^

123;
4_273;
42_73;
427_3;
-34;


//  ^ ^ ^ ^ Integers: Type Modifiers ^ ^ ^ ^

0xF5;
0o12;
0b110;


//   ^ ^ ^ ^  Integers: Unsigned  ^ ^ ^ ^ 

11u8;
12u16;
13u32;
14u64;
15u128;
16usize;

//   ^ ^ ^ ^ Integers: Signed ^ ^ ^ ^

// ?

// floating point
// bools
// chars

12;
45.6;
"Hi";
true;
(1u32, 2u32, 4.3f32, 5.4f32);

//   & & & & & & Operators & & & & & &  

3.4 + 5.6;
true && false;

//   & & & & & & String Formatting & & & & & &   // 
// - Note the printing is MACRO based, :o/b/x for diffrent radixes //

println!("first: {}", my_tuple.0);

//   & & & & & & Conditional Statements & & & & & &   // 

//   & & & & & & Loops & & & & & &   //

//
fn main()
{
    // literals:
    
    // tuples:
    
    let my_tuple = (1u32, 2u32, 3u32, 4.3f32, 5.4f32);     
    
    

    // incremental //
    print!("{0}, this is {1}. {1}, this is {0}", "Alice", "Bob")
    
    // no order significance //
    println!("{subject} {verb} {object}", object = "the lazy dog", subject = "the quick brown fox", verb = "jumps over")

    println!("decimal: {}", 1234);
    println!("hexa: {:x}", 1234);
    println("octal: {:o}", 1234);
    println!("binary: {:b}", 1234);
    // paragraphing - justifying and adding zeros //

    println!("{number:>5}", number = 2); // right justify by 4 places (the last taken by the 2) //
    // left justifying is like printing regularly //
    println!("{number:0>5}", number = 2); // justify right with zeros //
    println!("{number:0<5}", number = 2); // justify left with zeros //
    println!("{number:0>width$}", number = 2, width = 5);

    // immutable variables //
    
    let number : f64 = 2.0;
    let width : usize = 5;
    println!("{number:>width$}"); // right justify with zeros and limit decimal places //

    let fir_var = true;
    let fir_var: bool = false; // type annotation
    let sec_var: f64 = 4.5; // type annotation
    let third_var = 5i32; // sufixed type annotation
    let fist_tuple = ("hi", "bye");
    
    // mutable variables //

    let mut fort_var: f64 = 67.23;
    fort_var = 25.47f64; 

    // operators and literals

    println!("{}", 1u32 + 2);
    println!("{}", 1i32 - 2);
    println!("{}", 1e4);
    println!("{}", 1e5 - 1e4);

} 